.github/workflows/build-deploy-frontend-dev.yml

```
@@ -87,6 +185,6 @@ jobs:
             --bucket ${{ env.DEV_ARTIFACT_BUCKET }} \
             --key frontend.zip \
             --body frontend.zip \
-            --metadata "repository=$GITHUB_REPOSITORY,commitsha=$GITHUB_SHA,committag=$GIT_TAG,commitmessage=$COMMIT_MSG"`
+            --metadata '${{ toJson(fromJson(needs.pr-data.outputs.data)) }}'`
```

Is the not json => json conversion here just for condensing the string length?

```suggestion
            --metadata '${{ needs.pr-data.outputs.data }}'`
```

---

ref. dedupe: splitting the js out to a separate file is a nightmare with github script - you're welcome to give it a go, though!.

I guess we could split this job out to a separate reusable workflow, but the scope access can get a bit weird. I think a full dedupe is necessary on these workflows - they are on my hit list for future.

ref. json -> json, agreed, it looks janky as hell. the issue is that the output from the response isn't 'proper' json. The script action is weird - it returns a javascript object, which is 'kinda' json'y, but doesn't properly parse when passed on to aws. this to/from shuffle sorts it out and presents it to AWS in a way it can cope with. I did try it without, but it's super flaky